app/pages/lab/workflow-viewer/index.cjsx

```diff
@@ -0,0 +1,51 @@
+React = require 'react'
+apiClient = require '../../../api/client'
+PromiseRenderer = require '../../../components/promise-renderer'
+ChangeListener = require '../../../components/change-listener'
+WorkflowNodes = require './workflow'
+{Navigation} = require '@edpaget/react-router'
+
+WorkflowVis = React.createClass
+  displayName: 'WorkflowVis'
+
+  mixins: [Navigation]
+
+  getInitialState: ->
+    # make a new jsPlumb instance each time component is mounted
+    jp: jsPlumb.getInstance()
+
+  onSort: ->
+    @refs['workflowNodes'].sortTasks()
+
+  workflowLink: ->
+    [owner, name] = @props.project.slug.split('/')
+    viewParams = {owner, name}
+    viewQuery = workflow: @props.workflow.id, reload: 0
+    @makeHref 'project-classify', viewParams, viewQuery
+
+  render: ->
+    <div>
+      <div style={pointerEvents: 'all'}>
+        <span className='standard-button' onClick={@onSort}>Sort Task Nodes</span>
+      </div>
+      <hr />
+      <WorkflowNodes jp={@state.jp} workflow={@props.workflow} ref='workflowNodes' />
+      <hr />
+      <div style={pointerEvents: 'all'}>
+        <a href={@workflowLink()} className="standard-button" target="from-lab" onClick={@handleViewClick}>Test this workflow</a>
+      </div>
+    </div>
+
+module.exports = React.createClass
+  displayName: 'WorkflowVisPageWrapper'
+
+  getDefaultProps: ->
+    params:
+      workflowID: ''
```

Comment:
I don't think this means much here, considering the page requires a workflow ID to do anything. Maybe using `React.PropTypes.shape` with a required param of `workflowID`?